worker/src/main/java/gov/cms/ab2d/worker/processor/JobPreProcessorImpl.java

```diff
@@ -0,0 +1,43 @@
+package gov.cms.ab2d.worker.processor;
+
+import gov.cms.ab2d.common.model.Job;
+import gov.cms.ab2d.common.repository.JobRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import static gov.cms.ab2d.common.model.JobStatus.IN_PROGRESS;
+import static gov.cms.ab2d.common.model.JobStatus.SUBMITTED;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class JobPreProcessorImpl implements JobPreProcessor {
+
+    private final JobRepository jobRepository;
+
+    @Override
+    @Transactional(propagation = Propagation.REQUIRES_NEW, isolation = Isolation.SERIALIZABLE)
+    public Job preprocess(String jobUuid) {
+
+        final Job job = jobRepository.findByJobUuid(jobUuid);
+        if (job == null) {
+            log.error("Job was not found");
+            throw new IllegalArgumentException("Job " + jobUuid + " was not found");
```

Comment:
I think this should actually be a ResourceNotFoundException found in the common module instead, IllegalArgumentException could conflict with other exceptions thrown from the JDK or other libraries.

Reply:
I do not know if I agree with this comment.
If fact, in `Effective Java -  2nd ed`, *Joshua Bloch* recommends using the it.  _Item 60 - Favor the user of standard exceptions_  (page 248-9).
https://github.com/tatsuyaoiw/effective-java/blob/master/chapter-9.md

Over here, the scenario is that the method was provided a parameter which was bad data and hence inappropriate. 

Also, it is fine to call it is a Resource in the context of a Rest API so that the exception handling can handle the exception as needed when a resource is not found. However, since the worker does not provide a rest endpoint, and is mostly batch oriented, I think it is better to keep it generic and not call it a resource in this context.